icu4c/source/i18n/hebrwcal.cpp

```diff
@@ -423,7 +423,15 @@ int32_t HebrewCalendar::startOfYear(int32_t year, UErrorCode &status)
             // Prevents 382-day years.
             day += 1;
         }
-        CalendarCache::put(&gCache, year, day, status);
+        if (day > INT32_MAX || day < INT32_MIN) {
+            status = U_ILLEGAL_ARGUMENT_ERROR;
+            return 0;
+        }
+        CalendarCache::put(&gCache, year, static_cast<int32_t>(day), status);
+    }
+    if (day > INT32_MAX || day < INT32_MIN) {
+        status = U_ILLEGAL_ARGUMENT_ERROR;
+        return 0;
```

Comment:
You have this same check twice in the same function, inside and outside in `if` block.  Can you consolidate?  And if not, especially if you've got this same check in a bunch of other places, is it worth making this a macro or an inline function for readability?

Reply:
ok, since I check that before I put into cache I think this one is not necessary